src/codegen_call_native.cpp

```
@@ -199,3 +260,226 @@ llvm::Value *CodeGen::emitTableDrivenNativeCall(
+//
+// This handles calls to functions declared with @native("libname") that are
+// NOT covered by the hardcoded stdlib dispatch tables. It uses the signature
+// registry to derive the C calling convention from the Ry type annotations.
+
+namespace {
+
+// Infer the ReturnWrapping from the Ry return type name.
+// Returns {wrapping, out_param_type_name} where out_param_type_name is non-empty
+// only for ResultOutParam.
+std::pair<CodeGen::ReturnWrapping, std::string>
+inferReturnWrapping(const std::string &returnType) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)